k8s/README.md

```diff
@@ -0,0 +1,203 @@
+
+## Prerequisites
+
+* [Set up a Kubernetes Cluster](http://kubernetes.io/docs/getting-started-guides/)
+* [Installing `kubectl`](http://kubernetes.io/docs/user-guide/prereqs/)
+
+## Before you begin
+
+1. Select or create a [Cloud Platform Console project](https://console.cloud.google.com/project).
+
+2. [Enable billing](https://support.google.com/cloud/answer/6293499#enable-billing) for your project.
+
+3. Note the project ID, because you'll need it later.
+
+4. Install [cURL](https://curl.haxx.se/download.html) for testing purposes.
+
+5. [Enable Cloud Endpoints API](https://console.cloud.google.com/apis/api/endpoints.googleapis.com/overview)
+   for your project in the Google Cloud Endpoints page in the API Manager.
+   Ignore any prompt to create credentials.
+
+6. [Download the Google Cloud SDK](https://cloud.google.com/sdk/docs/quickstarts).
+
+## Configuring Endpoints
+
+To configure Endpoints, replace `YOUR-PROJECT-ID` with your own project ID in
+the [swagger.yaml](swagger.yaml) configuration file:
+    
+   ```
+   swagger: "2.0"
+   info:
+     description: "A simple Google Cloud Endpoints API example."
+     title: "Endpoints Example"
+     version: "1.0.0"
+   host: "YOUR-PROJECT-ID.appspot.com"
+   ```
+
+## Deploying the sample API config to Google Service Management
+
+To deploy the sample application:
+
+1. Invoke the following command:
+
+   ```
+   gcloud beta service-management deploy swagger.yaml
+   ```
+
+   The command returns several lines of information, including a line similar to the following:
+
+   ```
+   Service Configuration with version "2016-04-27R2" uploaded for service "YOUR-PROJECT-ID.appspot.com"
+   ```
+
+   Note that the version number that displayed will change when you deploy a new
+version of the API.
+
+2. Make a note of the service name and the service version because you'll need
+them later when you configure the container cluster for the API.
+
+## Deploying the sample API to the cluster
+
+To deploy to the cluster:
+
+1. Edit the Kubernetes configuration file,
+i.e. [esp_echo_http.yaml](esp_echo_http.yaml),
+replacing `SERVICE_NAME` and `SERVICE_VERSION` shown in the snippet below with
+the values returned when you deployed the API:
+
+   ```
+   containers:
+     - name: esp
+       image: b.gcr.io/endpoints/endpoints-runtime:0.3
+       command: [
+         "/usr/sbin/start_esp.py",
+         "-p", "8080",            # the port ESP listens on
+         "-a", "127.0.0.1:8081",  # the backend address
+         "-s", "SERVICE_NAME",
+         "-v", "SERVICE_VERSION",
+         "-k", "/etc/nginx/creds/service-account-creds.json",  # not needed for GKE
+       ]
+   ```
+
+   Note you also need to change the service type from LoadBalancer to NodePort
+   if you use [MiniKube](http://kubernetes.io/docs/getting-started-guides/minikube/)
+
+2. [Create your service account credentials](https://cloud.google.com/storage/docs/authentication#generating-a-private-key)
```

Comment:
2 and 3 Should be merged into a single Bullet point, and you should START that bullet point with "(Not necessary if you kubernetes cluster is on GKE)"